### PR TITLE
SONARFLEX-84 Feed metric 'executable_lines_data'

### PR DESCRIPTION
**Pull request by mpaladin**

### CODE REVIEW

File: flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java [hunk 1]
```diff
@@ -31,11 +33,12 @@
   private final int numberOfClasses;
   private final int numberOfFunctions;
   private final FileLinesVisitor fileLinesVisitor = new FileLinesVisitor();
+  private final String executableLines;
 
   public FileMetrics(FlexVisitorContext context) {
     AstNode rootTree = context.rootTree();
     Objects.requireNonNull(rootTree, "Cannot compute metrics without a root tree");
-    numberOfStatements = rootTree.getDescendants(
+    List<AstNode> descendants = rootTree.getDescendants(
```
Comment (pynicolas) [inline flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java]: I believe that `statements` would be a better name.

File: flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java [hunk 2]
```diff
@@ -50,7 +53,19 @@ public FileMetrics(FlexVisitorContext context) {
...
       FlexGrammar.THROW_STATEMENT,
       FlexGrammar.TRY_STATEMENT,
-      FlexGrammar.EMPTY_STATEMENT).size();
+      FlexGrammar.EMPTY_STATEMENT);
+
+    Set<Integer> alreadyMarked = new HashSet<>();
+    StringBuilder sb = new StringBuilder();
+    for (AstNode descendant : descendants) {
+      int line = descendant.getTokenLine();
+      if (alreadyMarked.add(line)) {
```
Comment (pynicolas) [inline flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java]: This is not tested.

File: flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java (hunk 1 above)
Comment (mpaladin) [reply flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java]: @pynicolas renamed in https://github.com/SonarSource/sonar-flex/pull/73/commits/80e609d8038bb9e8ced8130e7cdb4b6d1b3adf24.

File: flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java (hunk 2 above)
Comment (mpaladin) [reply flex-squid/src/main/java/org/sonar/flex/metrics/FileMetrics.java]: @pynicolas test added in https://github.com/SonarSource/sonar-flex/pull/73/commits/80e609d8038bb9e8ced8130e7cdb4b6d1b3adf24.

**Comment by mpaladin** [comment]
@pynicolas IT assertion added in https://github.com/SonarSource/sonar-flex/pull/73/commits/cd9931c18cdcf064203bd1a314c9f00371757738.

**Pull request merged**